gollm/gemini.go

```diff
@@ -67,24 +66,18 @@ var _ Client = &GeminiClient{}
 
 // ListModels lists the models available in the Gemini API.
 func (c *GeminiClient) ListModels(ctx context.Context) (modelNames []string, err error) {
-	models := c.client.ListModels(ctx)
-
-	for {
-		m, err := models.Next()
+	for model, err := range c.client.Models.All(ctx) {
 		if err != nil {
-			if err == iterator.Done {
-				break
-			}
-			return nil, err
+			return nil, fmt.Errorf("error listing models: %w", err)
 		}
-		modelNames = append(modelNames, strings.TrimPrefix(m.Name, "models/"))
+		modelNames = append(modelNames, strings.TrimPrefix(model.Name, "models/"))
```

Comment:
Might be interesting to see what other metadata we have here (re our discussion to figure out whether models support function calling!)

Reply:
Good pt. Quick look https://pkg.go.dev/google.golang.org/genai#Model shows `SupportedActions` as something interesting. Will look into it.

```
// List of actions that are supported by the model.
	SupportedActions [][string](https://pkg.go.dev/builtin#string) `json:"supportedActions,omitempty"`
```